pkg/process/process.go

```diff
@@ -35,6 +36,45 @@ const (
 	trackableFalse = false
 )
 
+// New creates a new Process object.
+func New(
+	kebClient *keb.Client,
+	edpClient *edp.Client,
+	edpCollector collector.CollectorSender,
+	configProvider runtime.ConfigProvider,
+	publicCloudSpecs *config.PublicCloudSpecs,
+	scrapeInterval time.Duration,
+	workerPoolSize int,
+	logger *zap.SugaredLogger,
+) (*Process, error) {
+	switch {
+	case logger == nil,
+		configProvider == nil,
+		publicCloudSpecs == nil,
+		edpCollector == nil,
+		edpClient == nil,
+		kebClient == nil:
+		return nil, fmt.Errorf("missing required parameter")
+	}
```

Comment:
Is there a reason to not consider `scrapeInterval` and `workerPoolSize` also as required parameters?